### PR TITLE
[civ2][flakiness/6] support run tests with multiple size

### PR DESCRIPTION
**Pull request by can-anyscale**
Support running tests with different type of test size, this is to be feature-parity with civ01; they are running tests small & medium tests in medium size machine and large tests in large machine

Test:
- CI

### CODE REVIEW

**Comment by can-anyscale** [comment]
tada

Screenshot 2023-07-28 at 10 06 15 AM

File: .buildkite/pipeline.build_core.yml [hunk 1]
```diff
@@ -1,11 +1,23 @@
 #ci:group=:book: core tests (civ2)
 
-- label: ":python:"
+- label: ":python: small & medium"
   conditions: ["RAY_CI_PYTHON_AFFECTED"]
   instance_size: medium
   parallelism: 3
+  parallelism: 3
```
Comment (aslonnie) [inline .buildkite/pipeline.build_core.yml:7]: why there are two `parallelism` here now?

**Review by aslonnie — COMMENTED** [review]
why do we want to support running in multiple sizes?

**Comment by can-anyscale** [comment]
> why do we want to support running in multiple sizes?

just to be feature-parity with civ01; they are running tests small & medium tests in medium size machine and large tests in large machine

File: ci/ray_ci/core.tests.yml [hunk 2]
```diff
@@ -5,10 +5,8 @@ flaky_tests:
   - //python/ray/tests:test_placement_group_5
   - //python/ray/tests:test_runtime_env_2
   - //python/ray/tests:test_gcs_fault_tolerance
-  - //python/ray/tests:test_gcs_ha_e2e
```
Comment (can-anyscale) [inline ci/ray_ci/core.tests.yml:8]: remove since we now filter this through the except query tin runner.py

File: .buildkite/pipeline.build.yml [hunk 3]
```diff
@@ -202,12 +202,18 @@
...
-  parallelism: 3
   commands:
     - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
     - DL=1 ./ci/env/install-dependencies.sh
     - ./ci/env/env_info.sh
-    - ./ci/ci.sh test_large
+    - bazel test --config=ci $(./ci/run/bazel_export_options) --test_env=CONDA_EXE --test_env=CONDA_PYTHON_EXE
+      --test_env=CONDA_SHLVL --test_env=CONDA_PREFIX --test_env=CONDA_DEFAULT_ENV --test_env=CONDA_PROMPT_MODIFIER
+      --test_env=CI --test_tag_filters="large_size_python_tests_shard_0,large_size_python_tests_shard_1,large_size_python_tests_shard_2"
+      -- $(yq -r .flaky_tests[] ci/v2/test/core.tests.yml)
```
Comment (aslonnie) [inline .buildkite/pipeline.build.yml:212]: is it true that at the end of this series of changes, the `yq` will go away where we will be parsing these yml files with `ray_ci` python script?

**Review by aslonnie — APPROVED** [review]

File: ci/ray_ci/runner.py [hunk 4]
```diff
@@ -77,24 +86,32 @@ def _chunk_into_n(list: List[str], n: int):
...
     )
 
-    return f"{team_query} intersect ({size_query})"
+    return f"({team_query} intersect ({size_query})) except ({except_query})"
 
 
-def _get_all_test_targets(targets: str, team: str, yaml_dir: str) -> List[str]:
+def _get_all_test_targets(
+    targets: str, team: str, size: str, yaml_dir: str
+) -> List[str]:
```
Comment (aslonnie) [inline ci/ray_ci/runner.py:107]: I feel that the filtering logic in these functions are getting a bit complicated now..

File: .buildkite/pipeline.build.yml (hunk 3 above)
Comment (can-anyscale) [reply .buildkite/pipeline.build.yml:212]: that's correct yes, getting there

File: ci/ray_ci/runner.py (hunk 4 above)
Comment (can-anyscale) [reply ci/ray_ci/runner.py:107]: It's an artifact of  tag usages in civ01 that makes unification rather complex. I hope to clean this up along the way.

**Pull request merged**